src/main/scala/io/github/pauljamescleary/petstore/domain/pets/PetStatus.scala

```diff
@@ -1,20 +1,12 @@
 package io.github.pauljamescleary.petstore.domain.pets
+import enumeratum._
 
-sealed trait PetStatus extends Product with Serializable
-case object Available extends PetStatus
-case object Pending extends PetStatus
-case object Adopted extends PetStatus
+sealed trait PetStatus extends EnumEntry
 
-object PetStatus {
-  def apply(name: String): PetStatus = name match {
-    case "Available" => Available
-    case "Pending" => Pending
-    case "Adopted" => Adopted
-  }
+case object PetStatus extends Enum[PetStatus] with CirceEnum[PetStatus] {
+  case object Available extends PetStatus
+  case object Pending extends PetStatus
+  case object Adopted extends PetStatus
 
-  def nameOf(status: PetStatus): String = status match {
-    case Available => "Available"
-    case Pending => "Pending"
-    case Adopted => "Adopted"
-  }
+  val values = findValues
```

Comment:
what is the reason for `val values = ...`?  

Also, perhaps add an explicit type for `values`?

Reply:
I'm no expert in enumeratum, but I believe is boilerplate required for the macros to work. See [this comment in the code](https://github.com/lloydmeta/enumeratum/blob/master/enumeratum-core/src/main/scala/enumeratum/Enum.scala#L150-L156). There's [an issue around dealing with the boilerplate](https://github.com/lloydmeta/enumeratum/issues/141), but it's still open.

As for the adding in an explicit type to `values`, it's a `scala.collection.immutable.IndexedSeq[A]`. Would you prefer to be as explicit as possible or would you rather a plain `Seq[A]`?

Reply:
@ml10 the `values` doesn't have an explicit return type in the Enumeratum docs (just looked) so it is ok to leave it out.

Reply:
@martin-g  👍  ?